src/main/java/org/candlepin/subscriptions/capacity/PoolIngressController.java

```diff
@@ -75,13 +79,53 @@ public PoolIngressController(SubscriptionCapacityRepository repository,
         capacityRecordsDeleted = meterRegistry.counter("rhsm-subscriptions.capacity.records_deleted");
     }
 
+    @Transactional
+    @Timed("rhsm-subscriptions.subscription.ingress")
+    public void updateSubscriptionsForOrg(String orgId, List<CandlepinPool> pools) {
+        final List<String> subscriptionIds = pools.stream().map(CandlepinPool::getSubscriptionId)
+            .collect(Collectors.toList());
+        final Collection<Subscription> existingSubscriptionRecords = subscriptionRepository
+            .findByOwnerIdAndSubscriptionIdIn(orgId, subscriptionIds);
+
+        final Map<String, Subscription> skuToSubscription = new HashMap<>();
+        existingSubscriptionRecords
+            .forEach(subscription -> skuToSubscription.put(subscription.getSku(), subscription));
+
+        final Collection<Subscription> needsSave = new ArrayList<>();
+        final Collection<Subscription> needsDelete = new ArrayList<>();
+        pools.forEach(pool -> {
+            final String poolSku = pool.getProductId();
+            if (productWhitelist.productIdMatches(poolSku)) {
+                Subscription updatableSubscription = skuToSubscription.get(poolSku);
+                if (updatableSubscription == null) {
+                    updatableSubscription = new Subscription();
+                    updatableSubscription.setSku(poolSku);
+                }
+                else {
+                    skuToSubscription.remove(poolSku);
+                }
+                updatableSubscription.setSubscriptionId(pool.getSubscriptionId());
+                updatableSubscription.setStartDate(pool.getStartDate());
+                updatableSubscription.setEndDate(pool.getEndDate());
+                updatableSubscription.setOwnerId(orgId);
+                updatableSubscription.setQuantity(pool.getQuantity());
```

Comment:
FYI, sometimes this will be 1:1 with the subscription quantity, and sometimes not.

https://access.redhat.com/sites/default/files/attachments/subscriptionandentitlementaccounting-v1.1.pdf has some good examples...

For example:
- given `RH0103678 (Red Hat Enterprise Linux Server, Premium (1-2 sockets) (Up to 1 guest) with Smart Management)`, the pool quantity will be equal to the subscription quantity.
- given `RS00013​ (Red Hat Ceph Storage for Management Nodes, Premium (Up to 6 Physical Nodes)`, the pool quantity would be 6 when the subscription quantity is 1.

For simplicity, let's just leave subscription quantity untouched for the short-term.

Reply:
sounds good!

Reply:
Removed lines where the ingress and ingress test set the quantity.